src/login/logind-session-dbus.c

```diff
@@ -830,6 +831,12 @@ int session_send_create_reply(Session *s, sd_bus_error *error) {
         if (fifo_fd < 0)
                 return fifo_fd;
 
+        if (s->leader.fd >= 0) {
```

Comment:
please don't do this test here. instead just call session_watch_pidfd() and make it exit early if we don't have the pidfd. i.e. abstract the test away.